pkg/remote/aws/route_table_supplier.go

```diff
@@ -0,0 +1,133 @@
+package aws
+
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/cloudskiff/driftctl/pkg"
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/hashicorp/terraform/flatmap"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type RouteTableSupplier struct {
+	reader                        terraform.ResourceReader
+	defaultRouteTableDeserializer deserializer.CTYDeserializer
+	routeTableDeserializer        deserializer.CTYDeserializer
+	client                        ec2iface.EC2API
+	defaultRouteTableRunner       *terraform.ParallelResourceReader
+	routeTableRunner              *terraform.ParallelResourceReader
+}
+
+func NewRouteTableSupplier(runner *pkg.ParallelRunner, client ec2iface.EC2API) *RouteTableSupplier {
+	return &RouteTableSupplier{
+		terraform.Provider(terraform.AWS),
+		awsdeserializer.NewDefaultRouteTableDeserializer(),
+		awsdeserializer.NewRouteTableDeserializer(),
+		client,
+		terraform.NewParallelResourceReader(runner.SubRunner()),
+		terraform.NewParallelResourceReader(runner.SubRunner()),
+	}
+}
+
+func (s RouteTableSupplier) Resources() ([]resource.Resource, error) {
+
+	retrievedRouteTables, err := listRouteTables(s.client)
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	for _, routeTable := range retrievedRouteTables {
+		res := *routeTable
+		var isMain bool
+		for _, assoc := range res.Associations {
+			if assoc.Main != nil && *assoc.Main {
+				isMain = true
+				break
+			}
+		}
+		if isMain {
+			s.defaultRouteTableRunner.Run(func() (cty.Value, error) {
+				return s.readRouteTable(res, true)
+			})
+			continue
+		}
+		s.routeTableRunner.Run(func() (cty.Value, error) {
+			return s.readRouteTable(res, false)
+		})
```

Comment:
We should first run one of the runner and wait for it to end before running and waiting for the other one. Otherwise we could have race conditions.

Reply:
It does not seem to be a race condition as we are using two separated runners.